src/main/java/bio/terra/app/usermetrics/BardClient.java

```diff
@@ -56,4 +78,69 @@ public void logEvent(AuthenticatedUserRequest userReq, BardEvent event) {
       logger.warn("Error logging event {}", event.getEvent(), e);
     }
   }
+
+  /**
+   * Syncing a user is only needed when a new token is not already cached or the cache entry has
+   * expired.
+   *
+   * @param userReq - the AuthenticatedUserRequest that represents the current user.
+   */
+  boolean syncUser(AuthenticatedUserRequest userReq) {
+    boolean updatedEntry = false;
+    String key = userReq.getToken();
+    if (!bearerCache.containsKey(key)) {
+      if (syncProfile(userReq)) {
+        // only cache the key if the sync request was successful.
+        bearerCache.put(key, null);
+        updatedEntry = true;
+      }
+    }
```

Comment:
I think `computeIfAbsent()` is a more succinct way to express this, provided that the `boolean` result isn't needed. Since the only use of the result is in test, maybe there's another way to test this without requiring the boolean result?

Reply:
I'd normally agree (and first implementation used putIfAbsent), but I learned that both putIfAbsent and computeIfAbsent rely on the mapped value not being null. See https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/Map.html#computeIfAbsent(K,java.util.function.Function)  In our case, the mapped value is null.  
The underlying data structure is being used more as an expiring set instead of a map, hence the approach above.

Reply:
If you store the `key` as the value then you could use `computeIfAbsent()`. (I wouldn't use `putIfAbsent()` since that will make the `syncProfile()` call regardless of the result of `map.containsKey(key)`.)

Reply:
I think that's a reasonable alternative, but it trades on operational expense.  Would you trade the key for on a simple, non-null value?  For example, using an empty String would yield the same results and _might_ result in lower memory usage.

Reply:
`bearerCache.computeIfAbsent(key, key -> syncProfile(userReq) ? "" : null);`

Reply:
I went with
`bearerCache.computeIfAbsent(key, k -> syncProfile(userReq) ? "" : null);`

Reply:
Since a string is a java object its contents aren't copied on assignment, the cost of storing `""` or `key` is the same (since `key` is already being stored and won't be GC'd).